blackjack.py

```diff
@@ -2,137 +2,243 @@
+        self.cards = [GameManager.drawcards(self), GameManager.drawcards(self)]
+        self.setcount()
+            
     def check(self):
-        if self.dcount > 21:
+        if self.count > 21:
+            print('\nDealer\'s hand: ', self.cards)
             print('Dealer went over 21 and you won!')
-            restart()
+            player.bank += player.bid
+            self.restart()
+            
+    def hit(self):
+        while self.count <= 13:
+            self.cards.append(self.drawcards())
+            if isinstance(self.cards[-1][0], list) == True:
+                self.count += self.cards[-1][0][1]
+            else:
+                self.count += self.cards[-1][0]
 
-class Player():
+    def setcount(self):
+        if isinstance(self.cards[0][0], list) == True and isinstance(self.cards[1][0], list) == True:
+            self.count = self.cards[0][0][1] + self.cards[1][0][1]
+        elif isinstance(self.cards[0][0], list) == True and isinstance(self.cards[1][0], list) == False:
+            self.count = self.cards[0][0][1] + self.cards[1][0]
+        elif isinstance(self.cards[0][0], list) == False and isinstance(self.cards[1][0], list) == True:
+            self.count = self.cards[0][0] + self.cards[1][0][1]
+        else:
+            self.count = self.cards[0][0] + self.cards[1][0]
 
+class Player(GameManager):
+     
     def __init__(self):
-        self.pcard = [drawcards(), drawcards()]
-        self.pcount = self.pcard[0][0] + self.pcard[1][0]
-        self.pcard1 = []
-        self.pcard2 = []
+        self.cards = [GameManager.drawcards(self), GameManager.drawcards(self)]
+        self.bank = 1000
+        self.setcount() 
 
+    def check(self):
+        if self.count > 21:
+            print('\nYour hand: ', self.cards)
+            print('You went over 21 and lost!')
+            self.bank -= self.bid
+            self.bankrupt()
+            self.restart()
+            game.counter = 0
+       
+    def bet(self):
+        print('You currently have', '$' + str(self.bank), 'in your bank.')
+        while True:
+            self.bid = input('\nHow much would you like to bet?: ')
+            if self.bid.isdigit() == True:
+                self.bid = int(self.bid)
+                if self.bid == 0:
+                    print('You can\'t bet nothing')
+                elif self.bid > self.bank:
+                    print('Insufficient fund in bank')
+                elif self.bid <= self.bank:
+                    break
+            else:
+                print('Please enter a valid number')
+                
     def hit(self):
-        self.pcard.append(drawcards())
-        self.pcount += self.pcard[-1][0]
+        self.cards.append(self.drawcards())
+        if isinstance(self.cards[-1][0], list) == True:
+            self.count += self.cards[-1][0][1]
+        else:
+            self.count += self.cards[-1][0]
 
     def stand(self):
-        if self.pcount > dealer.dcount:
+        dealer.hit()

```

Comment:
A Player probably shouldn't depend on the dealer directly like this. This will complicate the logic even further. It should be that the GameManager should depend on a player and a dealer, and call appropriate methods on them.

Reply:
Also, when an object depends on another, you should pass that dependency to the object that uses it as an instance variable.

```python
class Car:
    def __init__(gas_pedal, break_peadl):
        self.gas_pedal = gas_pedal
        self.break_pedal = break_pedal

    def go(self):
        self.gas_pedal.pump()

    def stop(self):
        self.break_pedal.pump()

g = GasPedal()
b = BreakPedal()
c = Car(g, b)
c.go()
c.stop()
```

This pattern is called dependency inversion, and the act of passing the dependencies into an object is called dependency injection.